system-addon/content-src/components/TopSites/TopSiteForm.jsx

```diff
@@ -145,12 +147,12 @@ export class TopSiteForm extends React.PureComponent {
           <button className="cancel" type="button" onClick={this.onCancelButtonClick}>
             <FormattedMessage id="topsites_form_cancel_button" />
           </button>
-          {this.props.editMode &&
+          {editMode &&
             <button className="done save" type="submit" onClick={this.onSaveButtonClick}>
-              <FormattedMessage id="topsites_form_save_button" />
+              <FormattedMessage id={this.props.url ? "topsites_form_save_button" : "topsites_form_add_button"} />
             </button>
           }
-          {!this.props.editMode &&
+          {!editMode &&
             <button className="done add" type="submit" onClick={this.onAddButtonClick}>
               <FormattedMessage id="topsites_form_add_button" />
```

Comment:
Looks like these buttons could be cleaned up by combining the `!editMode` and `editMode` cases and using a shared click handler where `onAddButtonClick` and `onSaveButtonClick` has a bunch of shared code already.

Reply:
@rlr Some reason I thought we discused `TOP_SITES_PIN` vs `TOP_SITES_INSERT` where PIN without an index or index of -1 would mean INSERT ??

Reply:
@Mardak I *think* that was `TOP_SITES_ADD`+`TOP_SITES_EDIT` vs TOP_SITES_INSERT`? Or maybe it was `TOP_SITES_DROP`+`TOP_SITES_INSERT`? Gah, don't remember. But, the difference between those two is that pin replaces and insert potentially causes a shift.

Reply:
But hmmm, I guess if you are pinning, the slot should be available and insert would do the same thing. Yeah, I think they could be combined.

Reply:
Oh right. we use INSERT for dropping to a specific index. It was ADD vs INSERT

Reply:
Just to clarify, we need to keep both `TOP_SITES_PIN` and `TOP_SITES_INSERT` actions in general to allow for adding/editing to specific positions vs shifting over pinned items. Right now this Edit UI uses PIN for edit/Save and INSERT for Add. We can keep it like that.

There is maybe a event handling simplification by making the `TOP_SITES_PIN` without an index (or as -1) behave like `TOP_SITES_INSERT` without an index (i.e., insert at position 0). This makes `TopSitesFeed` a little more complex, but it would allow the component code to always just dispatch `{type: at.TOP_SITES_PIN, data: {site, index: this.props.index}}`